cloudbuild/nightly/scripts/python-scripts/create_cluster.py

```diff
@@ -0,0 +1,144 @@
+            'lifecycle_config': {'auto_delete_ttl': '3600s'},
+        }
+    }
+    try:
+        # Create the cluster.
+        operation = cluster_client.create_cluster(
+            request={'project_id': project_id, 'region': region, 'cluster': cluster}
+        )
+        result = operation.result()
+        logging.info(result)
+        return True
+    except Exception as _:
+        logging.info(f'Could not create cluster {cluster} in {region}')
+        return False
+
+
+def main(argv: Sequence[str]) -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--cluster_name',
+        dest='cluster_name',
+        help='Name of the cluster to be created.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--project_id',
+        dest='project_id',
+        help='Project Id in which the cluster is created.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--region_array_string',
+        dest='region_array_string',
+        help='Space separated string of regions.',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '--dataproc_image_version',
+        dest='dataproc_image_version',
+        help='The Dataproc Image version used to create the cluster.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--num_workers',
+        dest='num_workers',
+        help='Number of workers in the cluster.',
+        type=int,
+        required=True,
+    )
+    parser.add_argument(
+        '--initialisation_action_script_uri',
+        dest='initialisation_action_script_uri',
+        help='Link to the initialisation script for the cluster.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--region_saving_file',
+        dest='region_saving_file',
+        help='Location of the file to save the final location in which the cluster is created.',
+        type=str,
+        required=True
+    )
+
+    args = parser.parse_args(argv[1:])
+    # Providing the values.
+    cluster_name = args.cluster_name
+    region_array = args.region_array_string.split(' ')
+    dataproc_image_version = args.dataproc_image_version
+    num_workers = int(args.num_workers)
+    initialisation_action_script_uri = args.initialisation_action_script_uri
+    project_id = args.project_id
+    region_saving_file = args.region_saving_file
+
+    for region in region_array:
+        logging.info(f'Attempting cluster creation with region {region}')
+        is_cluster_created = create_cluster(project_id, region, cluster_name, num_workers,
+                                            dataproc_image_version,
+                                            initialisation_action_script_uri)
+        if is_cluster_created:
+            file = open(region_saving_file, 'w')
+            file.write(region)
+            file.close()
+            break
```

Comment:
Lets mention in comments what exactly is being done here. Elaborate on why we're creating clusters in separate regions (resource quota limitations), and what is our approach here to tackle that.